File: 01718.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
	string input, key;
	getline(cin, input);
	cin >> key;

	int index_key = 0;
	string result = "";
	for (int i = 0; i < input.length(); i++) {
		if (input[i] == ' ') result += ' ';
		else {
			char lock;
			if (input[i] > key[index_key]) {
				lock = input[i] - key[index_key];
				lock += 'a' - 1;
			}
			else if (input[i] == key[index_key])
				lock = 'z';
			else
				lock = 'z' - (key[index_key] - input[i]);
			result += lock;
		}

		index_key++;
		index_key %= key.length();
	}

	cout << result << endl;
}

File: 01718.py
s = input()
key = input()

index_key = 0
result = ""
for i in range(len(s)):
    if s[i] == ' ':
        result += ' '
    else:
        if s[i] > key[index_key]:
            lock = ord(s[i]) - ord(key[index_key])
            lock += ord('a') - 1
            lock = chr(lock)
        elif s[i] == key[index_key]:
            lock = 'z'
        else:
            lock = ord('z') - (ord(key[index_key]) - ord(s[i]))
            lock = chr(lock)
        result += lock

    index_key += 1
    index_key %= len(key)

print(result)
